### ISSUE TITLE
Integrate ES2015 as a module

### ISSUE THREAD

**Issue by bakura10** [ungap/custom-elements#14]
Hi :),

Thanks a lot for this awesome library. I just have a small question: we are targetting ES2015 browsers only, but use the library as a module.

Is the correct way to import it this way?

```
import '@ungap/custom-elements/es.js';
```

It was a bit unclear from the documentation. This seems to work but I am not 100% sure this is the correct way :)

**Comment by WebReflection** [comment]
if your bunder can resolve that, **yes**, that's the correct way, or better, the smallest, non-transpiled, minified version of this module.

**Comment by bakura10** [comment]
Thanks a lot for confirming @WebReflection . Just another extra question: while doing some performance benchmark of our project, I found that on Chrome the MutationObserver apparently kicks, depsite Chrome supporting all the custom elements feature. I was under the assumption that the polyfill would just execute for Safari (for built-in extend) and for very old browsers (for which I don't care), but apparently my assumption was incorrect?

**Comment by WebReflection** [comment]
no MO should kick on Chromium or Firefox ... and you should be able to verify that `customElements.define` is the native function.

**Comment by bakura10** [comment]
Strange. On Chromium, with the polyfill included I have this:

<img width="1277" alt="image" src="https://user-images.githubusercontent.com/1198915/196626855-1ba80ae0-e09a-42c4-b53f-886cbd280353.png">

Which seems to point to this:

<img width="937" alt="image" src="https://user-images.githubusercontent.com/1198915/196626917-55536148-8ccd-4bb3-a3fe-2618630159c5.png">

I bundle my code with ESBuild, I wonder if this is a side effect (but normally not, the detection should be done before creating the MO no ?)

**Comment by WebReflection** [comment]
open a console in an empty page and type this:

```js
const {observe} = MutationObserver.prototype;
MutationObserver.prototype.observe = function () {
  console.log('IT KICKED');
  return observe.apply(this, arguments);
};
import('https://unpkg.com/@ungap/custom-elements');
```

read **IT KICKED** only on Safari/WebKit browsers.

**Comment by bakura10** [comment]
It seems that ESBuild may introduce an issue here then. I can see the IT KICKED on Chrome on my project :/. I will check further.

**Comment by WebReflection** [comment]
if you use the ES version and you target only ES2015 browsers I don't understand why you have transpiled code ther (all for/of, as example)

**Comment by bakura10** [comment]
I import/export my deps in a single file using ESBuild:

```
/* Import */
import '@ungap/custom-elements';

/* Export */
export * as FocusTrap from 'focus-trap';
export {animate, timeline, stagger, inView, scroll} from 'motion';
export {default as Delegate} from 'ftdomdelegate';
```

ESBuild will then transpile some code. But for some reason, it seems that due to that, it may cause the polyfill to execute on Chrome. I'm trying to check why.

**Comment by WebReflection** [comment]
then it's not the es.js version .... try to `import '@ungap/custom-elements/es.js';` instead and see how that goes.
btw, no, the index.js which is the main module is indeed transpiled because it targets IE11 too.

**Comment by bakura10** [comment]
Same result. I have triple checked and my ESBuild is configured to ESNext only, no IE11. Are you thinking that because of the var ? If that's the case this is how ESbuild transforms things, this does not mean it targets IE11 :).

**Comment by WebReflection** [comment]
what you could do, is to manually log the result of `legacy` after this line, by changing it in the index.js of the module you installed via npm https://github.com/ungap/custom-elements/blob/master/index.js#L303

if that's *false*, it means ESBuild is broken.

**Comment by bakura10** [comment]
Main document. And legacy is indeed false! On Chrome, legacy is supposed to be false no? legacy should evaluate to true on Safari, isn't it?

**Comment by bakura10** [comment]
So it seems the problem comes from here: https://github.com/ungap/custom-elements/blob/master/index.js#L455

It is executed on my Chrome, so it indeed tries to polyfill custom built-in. It should not enter into the catch clause on Chrome, correct?

**Comment by bakura10** [comment]
Got it...

Here is what happened:

* I am creating a Shopify theme, and I just realized that when using Shopify CLI (the CLI tool to develop themes), when using their dev environment, the scripts are downloaded twice.
* Due to that, the polyfill executed twice, and the second time, the polyfill tried to register again the "extends-li" component to detect if the browser supports built-in, and of course it throws this exception, which entered into the catch clause and kaboom:

<img width="887" alt="image" src="https://user-images.githubusercontent.com/1198915/196633768-576ab394-4c03-49f0-a7b8-9be2dbbccd15.png">

Mystery solved. Thanks for your time!

**Comment by WebReflection** [comment]
this is kinda bad (running scripts twice) but I think I should indeed prevent that from happening ... meaning, if two projects embed the polyfill and ship these should **not break** ... I'll file a new issue, feel free to follow progress there.

**Comment by bakura10** [comment]
Yes, I will report the issue to Shopify as well, but indeed, I feel the polyfill should first check if the element exist or not :)

**Comment by WebReflection** [comment]
it's up and running. You should now be able to include as you like the polyfill and never see MO kicking on Chrome or Firefox, or Edge. https://github.com/ungap/custom-elements/issues/15

**Comment by bakura10** [comment]
Thanks, I can confirm the polyfill is not running anymore :)
